src/main/java/net/masterthought/cucumber/reducers/ReportFeatureWithRetestMarkingFlackyMerger.java

```diff
@@ -0,0 +1,101 @@
+package net.masterthought.cucumber.reducers;
+
+import static java.util.Arrays.stream;
+import static java.util.stream.Stream.concat;
+import static net.masterthought.cucumber.json.support.Status.FAILED;
+import static net.masterthought.cucumber.json.support.Status.PASSED;
+import static org.apache.commons.lang3.ArrayUtils.add;
+import static org.apache.commons.lang3.ArrayUtils.addAll;
+
+import net.masterthought.cucumber.json.Element;
+import net.masterthought.cucumber.json.Feature;
+import net.masterthought.cucumber.json.Hook;
+import net.masterthought.cucumber.json.Tag;
+import net.masterthought.cucumber.json.support.Resultsable;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Stream;
+
+/**
+ * Merge list of given features in the same way as ReportFeatureWithRetestMerger plus marking flacky tests
+ * (that were failed and after rerun are passed) not failing the last one and moves exception and attachments from
+ * previously failed scenario for better analyzing
+ * <p>
+ * Uses when need to generate a report with rerun results analyzing flacky tests.
+ */
+final class ReportFeatureWithRetestMarkingFlackyMerger extends ReportFeatureWithRetestMerger {
+
+    @Override
+    public boolean test(List<ReducingMethod> reducingMethods) {
+        return reducingMethods != null
+                && reducingMethods.contains(ReducingMethod.MERGE_FEATURES_WITH_RETEST_MARKING_FLACKY);
+    }
+
+    @Override
+    protected void replace(Feature feature, Element[] elements, int i, Element current, int indexOfPreviousResult,
```

Comment:
take a look about this change https://codecov.io/gh/damianszczepanik/cucumber-reporting/compare/e2bac56c24ae93f1b9160503c3271e136b9868cb...3826b3fdab78e2eb50828cf88675f89709be04cb/src/src/main/java/net/masterthought/cucumber/reducers/ReportFeatureWithRetestMerger.java